code-challenges.test.js

```diff
@@ -15,42 +15,174 @@
+           
+           for(let i = 1; i < (n - 2); i++) {
+               fibaNums.push(fibaNums[fibaNums.length-1] + fibaNums[fibaNums.length- 2])
+           } return (fibaNums)
+       
+   }
+    
+console.log(fibaSeq(fibLength1))
+console.log(fibaSeq(fibLength2))
+
+// Test result:
+//FAIL  ./code-challenges.test.js
+// ● Test suite failed to run
+
+// ReferenceError: fibLength1 is not defined
+
+//   55 |     
+//   56 |
+// > 57 | console.log(fibaSeq(fibLength1))
+//      |                     ^
+//   58 | console.log(fibaSeq(fibLength2))
+//   59 |
+//   60 | // Test result:
+
+//   at Object.fibLength1 (code-challenges.test.js:57:21)
+//   at TestScheduler.scheduleTests (node_modules/@jest/core/build/TestScheduler.js:317:13)
+
+// Test Suites: 1 failed, 1 total
+// Tests:       0 total
+// Snapshots:   0 total
+// Time:        0.381 s
+// Ran
+// End of test result
+// Not sure how to fix the code to make it past the test. I was stuck trying to fix it to make it work. so I proceeded to the next problem.
 
 
 // --------------------2) Create a function that takes in an array and returns a new array of only odd numbers sorted from least to greatest.
 
 // a) Create a test with expect statements for each of the variables provided.
 
+describe("oddNum", () => {
+    it("The odd numbers in the array", () => {
+      expect(fullArr1()).toEqual([-9, 7, 9, 199])
+      expect(fullArr2()).toEqual([-823, 7, 23])
+    }
+)
+})
+
 const fullArr1 = [4, 9, 0, "7", 8, true, "hey", 7, 199, -9, false, "hola"]
-// Expected output: [-9, 7, 9, 199]
+// // Expected output: [-9, 7, 9, 199]
 
 const fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
 // Expected output: [-823, 7, 23]
 
+//Test result:
+// Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.369 s
+// Ran all test suites.
 
 // b) Create the function that makes the test pass.
-
+ //pesudo code: create a function called sortOddNum 
+ //that takes in an array of different data types and returns an array of odd numbers sorted from least to the greatest
+ // filter the array for datatype numbers using filter
+ // filter the resulting array for odd numbers using modolo method
+ // arrange the resulting array of odd numbers from smallest to greatest value using sort method
+
+   const sortOddNum = (array) => {
+       let oddNum =[] // to initialize oddNum array
+       let onlyNumbers =[]  // to initialize onlyNumbers
+      return onlyNumbers = array.filter(Number)
+      return oddNum = onlyNumbers.filter() % 2 !== 0
+      return sortOddNum = (oddNum.sort)
+   }
+
+  console.log(sortOddNum(fullArr1))
+  console.log(sortOddNum(fullArr2))
+
+  //Test result:
+  // Test Suites: 1 failed, 1 total
+// Tests:       1 failed, 1 total
+// Snapshots:   0 total
+// Time:        0.395 s
+// Ran all test suites.
+// error Command failed with exit code 1.
```

Comment:
Good pseudo code! 

Check out the syllabus page on HOF's for the [syntax of Filter ](https://github.com/learn-academy-2022-charlie/Syllabus/blob/main/javascript/higher-order-functions.md)